spring-xd-dirt/src/main/java/org/springframework/xd/dirt/integration/bus/RabbitBusCleaner.java

```diff
@@ -0,0 +1,150 @@
+
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+
+/**
+ *
+ * @author Gary Russell
+ */
+public class RabbitBusCleaner implements BusCleaner {
+
+	final static Log logger = LogFactory.getLog(RabbitBusCleaner.class);
+
+	@Override
+	public List<String> clean(String stream) {
+		return clean("http://localhost:15672", "guest", "guest", "/", "xdbus.", stream);
+	}
+
+	@SuppressWarnings("unchecked")
+	public List<String> clean(String adminUri, String user, String pw, String vhost,
+			String busPrefix, String stream) {
+		List<String> results = new ArrayList<>();
+		RestTemplate restTemplate = buildRestTemplate(adminUri, user, pw);
+		int n = 0;
+		while (true) {
+			String queueName = busPrefix + stream + "." + n++;
+			URI uri = UriComponentsBuilder.fromUriString(adminUri + "/api").pathSegment("queues", "{vhost}", "{stream}")
+					.buildAndExpand(vhost, queueName).encode().toUri();
+			try {
+				Map<String, Object> queue = restTemplate.getForObject(uri, Map.class);
+				if (queue.get("consumers") != Integer.valueOf(0)) {
+					throw new RabbitAdminException("Queue " + queueName + " is in use");
+				}
+				results.add(queueName);
+				queueName += ".dlq";
+				uri = UriComponentsBuilder.fromUriString(adminUri + "/api").pathSegment("queues", "{vhost}", "{stream}")
+						.buildAndExpand(vhost, queueName).encode().toUri();
+				try {
+					queue = restTemplate.getForObject(uri, Map.class);
+					if (queue.get("consumers") != Integer.valueOf(0)) {
+						throw new RabbitAdminException("Queue " + queueName + " is in use");
+					}
+					results.add(queueName);
+				}
+				catch (HttpClientErrorException e) {
+					if (e.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
+						continue; // DLQs are not mandatory
+					}
+					throw new RabbitAdminException("Failed to lookup queue", e);
+				}
+			}
+			catch (HttpClientErrorException e) {
+				if (e.getStatusCode().equals(HttpStatus.NOT_FOUND)) {
+					break; // No more for this stream
+				}
+				throw new RabbitAdminException("Failed to lookup queue", e);
+			}
+		}
+		for (String queueName : results) {
+			URI uri = UriComponentsBuilder.fromUriString(adminUri + "/api").pathSegment("queues", "{vhost}", "{stream}")
+					.buildAndExpand(vhost, queueName).encode().toUri();
+			restTemplate.delete(uri);
+			if (logger.isDebugEnabled()) {
+				logger.debug("deleted queue: " + queueName);
+			}
+		}
+		return results;
+	}
+
+	public static RestTemplate buildRestTemplate(String adminUri, String user, String password) {
```

Comment:
That's how I had it originally, but I had to move the test to xd-shell because dirt no longer has rabbit on the cp; I guess I could put the test in the same package there, but it makes me feel a little queasy using a dirt package there.